membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -229,6 +224,28 @@ object AccountDetails {
   }
 }
 
+class FilterPlans(subscription: Subscription, catalog: Catalog)(implicit val logPrefix: LogPrefix) extends SafeLogging {
+
+  private val sortedPlans = subscription.ratePlans
+    .filter(_.product(catalog) match {
+      case _: Product.ContentSubscription => true
+      case Product.UnknownProduct => false
+      case Product.Membership => true
+      case Product.GuardianPatron => true
+      case Product.Contribution => true
+      case Product.Discounts => false
```

Comment:
discounts used to be filtered out in the high level reads, now they must be removed before we write the plans out to the client